analyses/cell-type-ewings/exploratory_analysis/annotation_notebooks/SCPCS000752/SCPCL001114_tumor-gene-expression-exploration.Rmd

```diff
@@ -0,0 +1,576 @@
+
+```{r}
+# plot distribution of ucell scores for both samples 
+ref_ucell <- ref_ucell |> 
+  as.data.frame() |> 
+  tibble::rownames_to_column("barcodes") |> 
+  dplyr::mutate(sample = "SCPCL000822")
+  
+ucell_df <- new_ucell |> 
+  as.data.frame() |> 
+  tibble::rownames_to_column("barcodes") |> 
+  dplyr::mutate(sample = "SCPCL001114") |> 
+  dplyr::bind_rows(ref_ucell) |> 
+  tidyr::pivot_longer(ends_with("UCell"), 
+                      names_to = "gene_list", 
+                      values_to = "signature_score") |> 
+  dplyr::mutate(
+    gene_list = stringr::word(gene_list, 1, sep = "_")
+  )
+```
+
+```{r}
+ggplot(ucell_df, aes(x = signature_score, colour = sample)) +
+  geom_density() +
+  facet_grid(rows = vars(gene_list),
+             scales = "free_y")
+```
+
+It looks like there is a bimodal distribution for `SCPCL000822` and not for `SCPCL001114` which is consistent with all the other marker gene methods we have looked at. 
+However, here we see that the signature scores for `SCPCL001114` tend to fall in the middle of the upper and lower distribution for `SCPCL000822`. 
+Because of this I don't think using `UCell` for this library would be very helpful. 
+
+### Consensus between `AUCell` and marker genes 
+
+We already saw that all cells called as tumor in `AUCell` were also called as tumor using the marker gene cutoff from `SCPCL000822`. 
+However, there are a group of normal cells that are called as tumor using marker genes but normal using `AUCell`. 
+Here we will take the consensus between both methods. 
+We will define any cells that are tumor in both as `Tumor`, normal in both as `Normal`, and any cells that are called differently in both as `Ambiguous`. 
+
+
+```{r}
+# add new column with consensus to classification df 
+classification_df <- classification_df |> 
+  dplyr::mutate(consensus_classification = dplyr::case_when(
+    ref_marker_gene_classification == "Tumor" & auc_classification == "Tumor" ~ "Tumor",
+    ref_marker_gene_classification == "Normal" & auc_classification == "Normal" ~ "Normal",
+    .default = "Ambiguous"
+  ))
+```
+
+## Validate classifications from all methods 
+
+First let's look at which cells are called tumor or normal on the UMAP across all classification methods. 
+
+```{r}
+method_levels <- c(
+  "1 marker gene" = "one_marker_gene_classification",
+  "2 marker genes" = "two_marker_gene_classification",
+  "Ref marker gene" = "ref_marker_gene_classification",
+  "AUCell" = "auc_classification", 
+  "Consensus" = "consensus_classification"
+)
+
+classification_df |> 
+  tidyr::pivot_longer(
+    cols = ends_with("classification"),
+    names_to = "method",
+    values_to = "classification"
+  ) |> 
+  dplyr::mutate(
+    method = forcats::fct_recode(method, !!!method_levels)) |> 
+  ggplot(aes(x = UMAP1, y = UMAP2, color = classification)) +
+  geom_point(size = 0.1, alpha = 0.5) +
```

Comment:
Two things here: I am not a fan of piping into ggplot and I woudl rather see you create a new df first.
But the more important thing is that I think it is worth throwing a `forcats::fct_reorder("Consensus", after = Inf)` to move the consensus to the end here (and everywhere else you do this transformation). It feels strange to have it be somewhere in the middle